--- a/src/module/imports.rs
+++ b/src/module/imports.rs
@@ -320,8 +320,8 @@ impl Emit for ModuleImports {
                         wasm_encoder::EntityType::Memory(wasm_encoder::MemoryType {
                             minimum: mem.initial,
                             maximum: mem.maximum,
-                            memory64: false,
+                            memory64: mem.memory64,
                             shared: mem.shared,
-                            page_size_log2: None,
+                            page_size_log2: mem.page_size_log2,
                         })
                     }
